### ISSUE TITLE
Expose git.publish as a command through Git API

### ISSUE THREAD

**Issue by nturinski** [microsoft/vscode#109977]
The `git.publish` command would be very helpful to have in the API itself.  Right now, push is very limited in that everything needs to already be set up for the user before we can push, but publish has a really great UI experience to help the user with everything.

**Comment by joaomoreno** [comment]
Have you tried to programmatically execute the `git.publish` command?

**Comment by nturinski** [comment]
Yes, but I haven't been able to catch any errors (if the repo you try to publish has no commits, for instance).

On a side-note, it would be nice if the `publishRepository` didn't await the information window to view the repo in GitHub since it's easy for a user to miss, but if used as part of a step, won't proceed to the next step until a user clicks something (either view or X)

**Comment by joaomoreno** [comment]
> Yes, but I haven't been able to catch any errors (if the repo you try to publish has no commits, for instance).

This is strange, when wrapping with `try { await executeCommand... } catch (err) {...}`, it never fails?

**Comment by nturinski** [comment]
It does fail, but for some reason, the error doesn't bubble up to my catch and gets thrown as a system modal error instead:

![image](https://user-images.githubusercontent.com/5290572/98583933-03cc3080-227a-11eb-84b0-8bd6c509501d.png)


git console:
![image](https://user-images.githubusercontent.com/5290572/98583969-0d559880-227a-11eb-86b0-2fb6f7251a2e.png)

My code:
```
  try {
        await commands.executeCommand('git.publish');
    } catch (err) {
        console.log(err);
    }
```

**Comment by joaomoreno** [comment]
And would you require this as a command, or is the actual API enough?

**Comment by nturinski** [comment]
I think an API would be good where I could pass the optional options of the repo that I want to publish.

Another unrelated note, but publish also defaults to the authenticated user so a user can't create a repo in an authorized organization through VS Code.

**Comment by joaomoreno** [comment]
> Another unrelated note, but publish also defaults to the authenticated user so a user can't create a repo in an authorized organization through VS Code.

https://github.com/microsoft/vscode/issues/98349
